plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java

```diff
@@ -182,6 +191,70 @@ static String bytesToStringConverter(Object value)
         return format("FROM_BASE64('%s')", Base64.getEncoder().encodeToString(slice.getBytes()));
     }
 
+    public static Field toField(String name, Type type)
+    {
+        if (type instanceof ArrayType) {
+            Type elementType = ((ArrayType) type).getElementType();
+            if (elementType instanceof RowType) {
+                return Field.newBuilder(name, StandardSQLTypeName.STRUCT, toFieldList((RowType) elementType)).setMode(Field.Mode.REPEATED).build();
+            }
+            return Field.newBuilder(name, toStandardSqlTypeName(elementType)).setMode(Field.Mode.REPEATED).build();
+        }
+        if (type instanceof RowType) {
+            return Field.of(name, StandardSQLTypeName.STRUCT, toFieldList((RowType) type));
+        }
+        return Field.of(name, toStandardSqlTypeName(type));
+    }
+
+    private static FieldList toFieldList(RowType rowType)
+    {
+        List<Field> fields = rowType.getFields().stream()
+                .map(field -> toField(field.getName().orElse("field" + rowType.getFields().indexOf(field)), field.getType()))
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
nit: static import and replace with `toImmutableList` since we aren't modifying it after return.